cmd/kubeadm/app/util/apiclient/idempotency.go

```diff
@@ -312,7 +314,10 @@ func PatchNodeOnce(client clientset.Interface, nodeName string, patchFn func(*v1
 		}
 
 		if _, err := client.CoreV1().Nodes().Patch(context.TODO(), n.Name, types.StrategicMergePatchType, patchBytes, metav1.PatchOptions{}); err != nil {
-			// TODO also check for timeouts
+			if apierrors.IsTimeout(err) {
+				fmt.Println("Temporarily unable to update node metadata due to timeout (will retry)")
+				return false, nil
+			}
```

Comment:
this seems fine. it is consistent with the IsConflict check.

do we have unit tests for IsConflict? if yes, i think unit tests should be added for IsTimeout accordingly.

Reply:
We currently have no boundary tests for these errors. Should i include them in this pr or issue another one to do this?

Reply:
I think this might be fine without unit tests and be an actual improvement, but i would like more reviewers to have a look, just in case.

/cc @SataQiu @pacoxu 
PTAL if you can

Reply:
thinking more about this...i think we should remove the fmt.Print* code from the function, even if it would hide the reason for a retry.

this code is considered backend, and backends should not print to stderr/out directly.

Reply:
a better solution to surface underlying errors is to make PatchNodeOnce accept a pointer to a local error e.g. `lastError`.

similarly to:
https://github.com/kubernetes/kubernetes/blob/cde45fb161c5a4bfa7cfe45dfd814f6cc95433f7/cmd/kubeadm/app/util/apiclient/idempotency.go#L340-L356

lastError can then be handled here:
https://github.com/kubernetes/kubernetes/blob/cde45fb161c5a4bfa7cfe45dfd814f6cc95433f7/cmd/kubeadm/app/util/apiclient/idempotency.go#L333

so that the last retry error surfaces to the user.